presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -718,6 +719,7 @@ public void evaluateIntermediate(int groupId, BlockBuilder output)
                 singleArrayBlockWriter.appendStructure(rowBlock.getBlock(i));
             }
             output.closeEntry();
+            blockBuilders.set(groupId, null);
```

Comment:
Shall we check if this is safe to do so? If yes, make a comment above this line?

Reply:
Yeah, I have deployed this to test clusters and this is safe. This is because
1) evaluateIntermediate is only called once spilling is invoked (never before that point) I.e. `SpillableHashAggregationBuilder.buildResult`/`spillToDisk()` occurs. Once we do this, aggregation Step is set to PARTIAL.
2) Then, we get the corresponding partial result for every aggregator by iterating over every groupId: https://fburl.com/kdl3hbhk. This groupId iterator is created in `hashSortedGroupIds` https://fburl.com/wrgy9wrc. As we can see in the line we are assigning each groupId an ordinal and it is impossible to have repeated groupIds in this list.

Will add a comment.